Gradual WIP changes

erelson opened the pull request:
Will push to my fork for now...

erelson committed:
Remove some but not all ps3 references

Update package.xml dep for binary drivers package

mikeferguson commented on fetch_binary_drivers/package.xml:
```diff
@@ -40,7 +40,7 @@
   <depend>boost</depend>
   <depend>curl</depend>
-  <depend>python</depend>
+  <depend>python3</depend>
```
If you want the same branch to support both noetic (and earlier) you can upgrade to package format 3 and use conditionals (see https://github.com/fetchrobotics/robot_controllers/blob/melodic-devel/robot_controllers/package.xml#L29)

erelson replied:
I had been planning on keeping melodic_devel for any melodic releases.  I'm wary of complexity (which in this case is anything new beyond the minimum needed 😬 ) given my limited time to test much.

erelson committed:
Update state_publisher -> robot_state_publisher

erelson committed:
Changes for compatibility with new drivers

Drivers primarily load some params from a json file which
end-users shouldn't need to modify. This file will live in
/opt/ros/noetic or the catkin workspace for 99% of usecases.

Add explicit setting for fetch teleop param.

erelson committed:
Point to a new drivers binary release (0.9.0)

cjds commented on fetch_system_config/root/lib/udev/rules.d/99-logitechF710.rules:
```diff
@@ -1,4 +0,0 @@
-KERNEL=="js?", SUBSYSTEM=="input", ATTRS{idVendor}=="046d", ATTRS{idProduct}=="c219", SYMLINK+="logitechF710"
```
No need for these anymore?

erelson replied:
We could walk back from this, but we've never shipped the Logitech F710 controller or similar with the research robots, and so, one less thing to support here, right?

erelson committed:
Respawn tuck_arm node on shutdown [OPEN-48]

JeffWilson7 commented on fetch_bringup/launch/include/teleop.launch.xml:
```diff
@@ -2,6 +2,7 @@
 
   <arg name="joy_device" default="/dev/fetch_joy"/>
   <arg name="ps4" default="false"/>
+  <arg name="is_fetch" default="false"/>
```
Doesn't look like this is used?

erelson replied:
Ahh, missed that in cleanup. Thanks!

At one point I had a bunch of added logic, before realizing why things weren't working at one point.  Why things weren't working: I ran freight.launch which sets /teleop/is_fetch to false, and then ran fetch.launch again later, and I could only teleop the base.... And that's why I now have the is_fetch = True line later in this launch file.

JeffWilson7 reviewed (approved):
Had one question...

erelson committed:
Fixup: Cleanup unused arg

cjds reviewed (approved):
LGTM

erelson committed:
Update maintainers

erelson committed:
but don't bump version manually

erelson commented:
Note to self for future (¯\_(ツ)_/¯): Squashing the commits wasn't helpful when I generated the changelogs.  (also changelog generation respects CATKIN IGNORE files; TIL)
